### PR TITLE
(2 Hours) As a guest I want to be able to save songs so I can remember good songs that played

### PR DESCRIPTION
**Pull request by mstojcevich**

### CODE REVIEW

**Comment by mstojcevich** [comment]
I plan on changing this to an ability to save the currently playing song instead of everything you've upvoted (since you probably already know a song if you upvoted it). This would require some larger changes (JS would need to know if the user is authenticated), so I'll do that later.

File: db/migrate/20161031012359_add_saved_playlist_id_to_user.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+class AddSavedPlaylistIdToUser < ActiveRecord::Migration[5.0]
```
Comment (Skovy) [inline db/migrate/20161031012359_add_saved_playlist_id_to_user.rb]: @mstojcevich do we need this column anymore?

File: app/models/user.rb [hunk 2]
```diff
@@ -42,4 +42,16 @@ def user_spotify_authenticated?
...
     @spotify ||= RSpotify::User.new(self.spotify_attributes)
   end
+
+  def add_to_saved(song)
+    track = RSpotify::Track.find(song.uri.sub!('spotify:track:', ''))
+
+    spotify.save_tracks!(Array(track))
+  end
+
+  def remove_from_saved(song)
```
Comment (Skovy) [inline app/models/user.rb]: @mstojcevich is this method used anywhere?

File: app/assets/javascripts/current_song.coffee [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      <div class="media-body">
+        <a href='#{@spotifyOpenURL()}' style='text-decoration: none' target='_blank'>
+          <h4 class='media-heading currently-playing__header'>
+            #{@name()}
+    """
+
+    # Put star button if user is authed with Spotify
+    if user_spotify_authed
+      html += """
+      <a class="btn" id="currently-playing__star" target="#">
```
Comment (Skovy) [inline app/assets/javascripts/current_song.coffee]: @mstojcevich is class `btn` intentional here?

**Pull request merged**